internal/controller/numaflowcontroller/numaflowcontroller_controller.go

```diff
@@ -239,6 +241,10 @@ func (r *NumaflowControllerReconciler) reconcile(
 				numaLogger.Warn("Dependent resources are not deleted yet, requeue after 5 seconds")
 				return ctrl.Result{Requeue: true}, nil
 			}
+			err := r.deleteChildren(ctx, controller.Namespace, controller.Name)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
```

Comment:
Because the manifests don't have an OwnerReference anymore, the deletion of the children is no longer automatic when NumaflowController is deleted, so we explicitly do it.